kratos/includes/amatrix_interface.h

```diff
@@ -438,6 +531,12 @@ template <typename TExpression1Type, typename TExpression2Type,
 AMatrix::VectorOuterProductExpression<TExpression1Type, TExpression2Type> outer_prod(
     AMatrix::MatrixExpression<TExpression1Type, TCategory1> const& First,
     AMatrix::MatrixExpression<TExpression2Type, TCategory2> const& Second) {
+    KRATOS_DEBUG_ERROR_IF(First.expression().size() != Second.expression().size())
```

Comment:
@jcotela i would say this is not correct. you can do other prod between a vector of size 2 and one of size 3 ... the result is a rectangular matrix